sdk/core/core/inc/az_span.h

```diff
@@ -5,17 +5,22 @@
 #define AZ_SPAN_H
 
 #include <az_assert.h>
-
+#include <az_option_byte.h>
 #include <az_static_assert.h>
-#include <stdint.h>
+
+#include <limits.h>
 #include <stdbool.h>
+#include <stdint.h>
 #include <string.h>
-#include <limits.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+#ifdef _MSC_VER
+#pragma warning(disable : 4204)
```

Comment:
pragma warning push at the top, and then warning pop at the bottom of the file?

Reply:
In the future, we may need something like `az_config.h`, `az_config_before.h`, `az_config_after.h`.

Reply:
I think it would be better to use macros, instead of including files like this.

You can refer to what we do in c++ stdlib headers for an example (we have a ton of warning suppression machinery, we don't have extern "c" { (for obvious reasons) but we do have an epilogue